### ISSUE TITLE
Use Layout with wrapped components

### ISSUE THREAD

**Issue by adumat** [react-toolbox/react-toolbox#1380]
I have an app container that use a `Layout` component, and inside i use a custom component like this
```
class NavigationBarContainer extends Component {

  constructor(props) {
    super(props);
    this.props = props;
  }

  toggleDrawerActive() {
    this.props.dispatch(navigationBarToggle());
  }

  toggleSubMenu(parentPosition) {
    this.props.dispatch(subMenuToggle({parentPosition}));
  }

  render() {
    let headerBackground = this.props.header.background;
    let headerAvatar = <Avatar><img src={this.props.header.avatar}/></Avatar>;
    let headerLabel = this.props.header.label;

    return (
      <NavDrawer active={this.props.isToggled} onOverlayClick={() => {
        this.toggleDrawerActive();
      }} {...this.props}>
        <Card className="navigationBar_header">
          <CardTitle
            avatar={headerAvatar}
            title={headerLabel}
          />
        </Card>
        <MenuList/>
      </NavDrawer>
    );
  }
}
```

it's work fine, but when i try to set props like `permanentAt` for responsive, the NavDrawer doesn't update its state.
if i use component directly on children of `Layout` component, NavDrawer works as expected.
is there any way for use custom components on Layout?

**Comment by jjstrat3** [comment]
I've been running into the same issue, and the only way I can see getting around it right now is using the `layoutFactory` directly from `Layout.js`.  The reason it doesn't work is the `layoutFactory` gets injected with the default `NavDrawer` from what I can tell and has an internal check with the `isComponentOfType` util function from this project which does a comparison.

I'm in the middle of working building up a custom layout component instance using that underlying factory function, so I'll report back whether it works or not, but I find it kind of clunky...surely there's a better way to do this?

**Comment by jjstrat3** [comment]
Nope.  Doesn't make a difference if you use the exported `layoutFactory` function.  Seems to not work if anything other than `themr` function from `react-css-themr` is the HOC.

**Comment by mjamado** [comment]
Is anybody looking into this?

**Comment by adumat** [comment]
yes, i'm waiting for a solution or workaround.
